LayoutTests/http/tests/workers/service/resources/indexeddb-cryptokey-put-get-worker.js

```diff
@@ -0,0 +1,153 @@
+                            reject(Error("Error in objectStore.get: " + JSON.stringify(e)));
+                        }
+                    }
+                } catch (e) {
+                    reject(Error("Error Exception Getting Key: " + JSON.stringify(e)));
+                }
+            }
+            request.onerror = function () {
+                reject(Error('ErrorOpeningDb'));
+            }
+            request.onupgradeneeded = function () {
+            }
+        }
+    );
+
+}
+
+async function testFetchedKeys(fetchedKeys) {
+    try {
+        for (let [key, value] of Object.entries(fetchedKeys)) {
+            let data = self.crypto.getRandomValues(new Uint8Array(100));
+            let iv = self.crypto.getRandomValues(new Uint8Array(16));
+            switch (key) {
+                case "AES-GCM":
+                case "AES-CBC":
+                    var ciphertext = new Uint8Array(await self.crypto.subtle.encrypt({name: key, iv: iv}, value, data));
+                    if (!(ciphertext.length > 0)) {
+                        return key;
+                    }
+                    break;
+                case "ECDSA":
+                    var signature = new Uint8Array(await self.crypto.subtle.sign({name: key, hash: "SHA-384"}, value.privateKey, data));
+                    if (!(signature.length > 0)) {
+                        return key;
+                    }
+                    break;
+                case "HMAC":
+                    signature = new Uint8Array(await self.crypto.subtle.sign(key, value, data));
+                    if (!(signature.length > 0)) {
+                        return key;
+                    }
+                    break;
+            }
+        }
+    } catch (e) {
+        return "Exception: " + e;
+    }
+    return "";
+}
+
+async function runTest() {
+    try {
+        let keys = await createKeys();
+        var fetchedKeys = {};
+        for (let [key, value] of Object.entries(keys)) {
+            await addValue(key, value);
+            fetchedKeys[key] = await getValue(key);
+        }
+        let check = await testFetchedKeys(fetchedKeys);
+        if (check.length > 0)
+            return new Response(null, {status: 200, statusText: "Test Failed: checkKeysFailed:  " + check});
+        return new Response(null, {status: 200, statusText: "Test Passed." + check});
+    } catch (e) {
+        return new Response(null, {status: 200, statusText: "Test Failed:Exception: " + JSON.stringify(e)});
+    }
+}
+
+async function cleanupDatabase() {
+    return new Promise(
+        function (resolve, reject) {
+            var request = indexedDB.deleteDatabase(dbName);
+            request.onsuccess = function () {
+                resolve("DB Deleted");
+            }
+            request.onerror = function () {
+                reject("Error Deleting DB");
+            }
+        }
+    );
+}
+
+self.addEventListener("activate", async () => {
+    await cleanupDatabase();
```

Comment:
Actually I think WebKitTestRunner cleans up data between test runs so this might not be needed.